### DISCUSSION TITLE
Raspberry pi 4,  cannot execute: required file not found

### DISCUSSION THREAD

**Discussion by adolfodeluca** [smartavionics/Cura#222] [General]
Hello everyone.
I download the the armhf version for raspberry pi 4, but I get this error when I execute

 ./Cura-4.20.12-mb-armhf.AppImage 
bash: ./Cura-4.20.12-mb-armhf.AppImage: cannot execute: required file not found

Before running it I installed libgles-dev

Thanks a lot from Argentina
Adolf

**Comment by smartavionics** [comment]
Hi, can you confirm that ./Cura-4.20.12-mb-armhf.AppImage is actually in the current directory and has been made executable (i.e. chmod +x ./Cura-4.20.12-mb-armhf.AppImage) ?

**Comment by adolfodeluca** [comment]
Hello, thanks for answer me.
Yes, the file have the correct permition

ls -ltr
total 283656
-rw-r--r-- 1 adeluca adeluca      1094 may  3 19:33 CuraConfiguration
-rwxr-xr-x 1 adeluca adeluca 219992324 may  3 20:02 Cura-4.20.12-mb-armhf.AppImage
-rw-rw-r-- 1 adeluca adeluca  70407813 may  3 20:11 mesa-armhf-22.2.5.tar.gz
drwxrwxr-x 3 adeluca adeluca      4096 may  3 20:23 opt
-rw-rw-r-- 1 adeluca adeluca     47111 may  3 20:46 README.html

or Cura...AppImage need to be into opt ?

**Comment by smartavionics** [comment]
No, the AppImage does not need to be in opt. What does it print when you do "file Cura-4.20.12-mb-armhf.AppImage", for me I see: 

Cura-4.20.12-mb-armhf.AppImage: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 2.6.32, BuildID[sha1]=06c350c8c3f914642421ec25cf1bc969ce45b84b, stripped

**Comment by adolfodeluca** [comment]
Hi, this...
adeluca@adeluca-rpi:~/Documents/Cura$ ls -ltr

total 283656
"-rw-r--r-- 1 adeluca adeluca      1094 may  3 19:33 CuraConfiguration
-rwxr-xr-x 1 adeluca adeluca 219992324 may  3 20:02 Cura-4.20.12-mb-armhf.AppImage
-rw-rw-r-- 1 adeluca adeluca  70407813 may  3 20:11 mesa-armhf-22.2.5.tar.gz
drwxrwxr-x 3 adeluca adeluca      4096 may  3 20:23 opt
-rw-rw-r-- 1 adeluca adeluca     47111 may  3 20:46 README.html"
adeluca@adeluca-rpi:~/Documents/Cura$ ./Cura-4.20.12-mb-armhf.AppImage 
bash: ./Cura-4.20.12-mb-armhf.AppImage: cannot execute: required file not found

**Comment by smartavionics** [comment]
What does it print when you do "file Cura-4.20.12-mb-armhf.AppImage", for me I see:

Cura-4.20.12-mb-armhf.AppImage: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 2.6.32, BuildID[sha1]=06c350c8c3f914642421ec25cf1bc969ce45b84b, stripped

**Comment by adolfodeluca** [comment]
Sorry , I see the same..

adeluca@adeluca-rpi:~/Documents/Cura$ file Cura-4.20.12-mb-armhf.AppImage 
Cura-4.20.12-mb-armhf.AppImage: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 2.6.32, BuildID[sha1]=06c350c8c3f914642421ec25cf1bc969ce45b84b, stripped

**Reply by smartavionics** [reply]
And /lib/ld-linux-armhf.so.3 does exist? What do you get if you do "ls /lib/ld-linux*" ?

**Comment by adolfodeluca** [comment]
only this

adeluca@adeluca-rpi:~/Documents/Cura$ ls -ltr /lib/ld-linux*
lrwxrwxrwx 1 root root 39 mar 16 05:44 /lib/ld-linux-aarch64.so.1 -> aarch64-linux-gnu/ld-linux-aarch64.so.1

**Comment by smartavionics** [comment]
Ah, you can see from that you are running a 64 bit kernel so you need the other AppImage...

https://github.com/smartavionics/Cura/releases/download/4.20.12/Cura-4.20.12-mb-aarch64.AppImage

**Comment by adolfodeluca** [comment]
ok, we have a new error, but is different 

adeluca@adeluca-rpi:~/Documents/Cura$ ./Cura-4.20.12-mb-aarch64.AppImage 
./Cura-4.20.12-mb-aarch64.AppImage: error while loading shared libraries: libz.so: cannot open shared object file: No such file or directory

**Comment by adolfodeluca** [comment]
just in case, a downloaded mesa for aarch64 and unzip again but, it was the same

**Comment by smartavionics** [comment]
What do you get when you do "ls -l /lib/aarch64-linux-gnu/libz*" ?

**Comment by adolfodeluca** [comment]
i get this 

adeluca@adeluca-rpi:~/Documents/Cura$ ls -l /lib/aarch64-linux-gnu/libz*
lrwxrwxrwx 1 root root     14 ene 11 14:02 /lib/aarch64-linux-gnu/libz.so.1 -> libz.so.1.2.13
-rw-r--r-- 1 root root 133280 ene 11 14:02 /lib/aarch64-linux-gnu/libz.so.1.2.13
lrwxrwxrwx 1 root root     16 mar  8 18:54 /lib/aarch64-linux-gnu/libzstd.so.1 -> libzstd.so.1.5.4
-rw-r--r-- 1 root root 657352 mar  8 18:54 /lib/aarch64-linux-gnu/libzstd.so.1.5.4

**Comment by smartavionics** [comment]
OK, please take a look at https://github.com/smartavionics/Cura/issues/157 and follow the instructions there which tells you how to use --appimage-mount and ldd to list all the libraries required. Please post the output here. Thanks.

**Comment by adolfodeluca** [comment]
Hi my friend, finally I resolved this issue installin this
sudo apt install zlib1g-dev

In stackOverflow I found a similar error, but with another program, and someone post that solution installing zlib1g-dev
https://stackoverflow.com/questions/21256866/libz-so-1-cannot-open-shared-object-file
Thanks a lot for guiad me.

this is the way.

Regards 
Adolfo

**Comment by smartavionics** [comment]
Great, glad to hear the problem is solved. Happy printing!
